unit/util/ieee_float.cpp

```diff
@@ -10,8 +10,91 @@ Author: Daniel Kroening, dkr@amazon.com
 
 #include <testing-utils/use_catch.h>
 
+#include <limits>
+
 TEST_CASE("Make an IEEE 754 one", "[core][util][ieee_float]")
 {
   auto spec = ieee_float_spect::single_precision();
   REQUIRE(ieee_floatt::one(spec) == 1);
 }
+
+TEST_CASE("round_to_integral", "[unit][util][ieee_float]")
+{
+  auto from_double = [](double d) -> ieee_float_valuet {
+    ieee_float_valuet v;
+    v.from_double(d);
+    return v;
+  };
+
+  auto round_to_integral =
```

Comment:
Could we also test for some small and large values halfway between integers,  previous below,  next above, x (e.g. 0.5, 0.49999999, 0.500000001) and cross that with the different rounding modes so check that  ties are actually resolved as expected?

'

Reply:
Added